pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -755,6 +780,82 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(ctx context.Context,
 	return reconcile.Result{}, nil
 }
 
+// validatePVCTopologyCompatibility checks if the existing PVC's topology annotation is compatible
+// with the volume's actual placement zone.
+func validatePVCTopologyCompatibility(ctx context.Context, pvc *v1.PersistentVolumeClaim,
+	volumeDatastoreURL string, topologyMgr commoncotypes.ControllerTopologyService,
+	vc *cnsvsphere.VirtualCenter) error {
+	log := logger.GetLogger(ctx)
+
+	// Check if PVC has topology annotation
+	topologyAnnotation, exists := pvc.Annotations[common.AnnVolumeAccessibleTopology]
+	if !exists || topologyAnnotation == "" {
+		// No topology annotation on PVC, skip validation
+		log.Debugf("PVC %s/%s has no topology annotation, skipping topology validation",
+			pvc.Namespace, pvc.Name)
+		return nil
+	}
+
+	// Parse PVC topology annotation
+	var pvcTopologySegments []map[string]string
+	err := json.Unmarshal([]byte(topologyAnnotation), &pvcTopologySegments)
+	if err != nil {
+		return fmt.Errorf("failed to parse topology annotation on PVC %s/%s: %v",
+			pvc.Namespace, pvc.Name, err)
+	}
+
+	// Get volume topology from datastore URL
+	volumeTopologySegments, err := topologyMgr.GetTopologyInfoFromNodes(ctx,
+		commoncotypes.WCPRetrieveTopologyInfoParams{
+			DatastoreURL:        volumeDatastoreURL,
+			StorageTopologyType: "zonal",
+			TopologyRequirement: nil,
+			Vc:                  vc,
+		})
+	if err != nil {
+		return fmt.Errorf("failed to get topology for volume datastore %s: %v", volumeDatastoreURL, err)
+	}
+
+	// Check if volume topology segments are compatible with PVC topology segments
+	if isTopologyCompatible(pvcTopologySegments, volumeTopologySegments) {
+		log.Infof("PVC %s/%s topology annotation is compatible with volume placement",
+			pvc.Namespace, pvc.Name)
+		return nil
+	}
+
+	// No compatible topology found
+	return fmt.Errorf("PVC %s/%s topology annotation %s is not compatible with volume placement in zones %+v",
+		pvc.Namespace, pvc.Name, topologyAnnotation, volumeTopologySegments)
+}
+
+// isTopologyCompatible checks if volume topology segments are compatible with PVC topology segments.
+// Returns true if all volume topology segments are satisfied by the PVC topology segments.
+func isTopologyCompatible(pvcTopologySegments, volumeTopologySegments []map[string]string) bool {
+	// For each volume topology segment, check if it exists in PVC topology segments
+	for _, volumeSegment := range volumeTopologySegments {
+		found := false
+		for _, pvcSegment := range pvcTopologySegments {
+			// Check if the volume segment satisfies the PVC's topology requirements
+			// This means all key-value pairs in the volume segment must match those in the PVC segment
```

Comment:
PVC can be requested in zone-1 or zone-2 or zone-3, but eventually getting created on either zone-1 or zone-2 or zone-3 provided using zonal storageclass.

if storage class without zonal requirement is used, and datastore selected is shared datastore then zones where datastore is accessible will be present on the pv node affinity rules.